orderly_web/constellation.py

```diff
@@ -0,0 +1,293 @@
+import os
+import tempfile
+import docker
+
+from PIL import Image
+
+import constellation
+import constellation.docker_util as docker_util
+
+from orderly_web.docker_helpers import docker_client
+
+
+def orderly_constellation(cfg):
+    redis = redis_container(cfg)
+    orderly = orderly_container(cfg, redis)
+    worker = worker_container(cfg, redis)
+    web = web_container(cfg)
+    containers = [redis, orderly, worker, web]
+
+    if cfg.proxy_enabled:
+        proxy = proxy_container(cfg, web)
+        containers.append(proxy)
+
+    obj = constellation.Constellation("orderly-web", cfg.container_prefix,
+                                      containers, cfg.network, cfg.volumes,
+                                      data=cfg, vault_config=cfg.vault)
+    return obj
+
+
+def redis_container(cfg):
+    redis_name = cfg.containers["redis"]
+    redis_mounts = [constellation.ConstellationMount("redis", "/data")]
+    redis_args = ["--appendonly", "yes"]
+    redis = constellation.ConstellationContainer(
+        redis_name, cfg.redis_ref, mounts=redis_mounts, args=redis_args,
+        configure=redis_configure)
+    return redis
+
+
+def redis_configure(container, cfg):
+    print("[redis] Waiting for redis to come up")
```

Comment:
This is coming from hint, we've seen failures when bringing up hintr before redis is ready to accept connections. I think we've not seen it in orderly yet as the redis data is much smaller but could be something we encounter in the future. Figured worthwhile to copy this over now instead of relying on redis starting up quickly enough to be available by the time we start the orderly.server container.